src/legendre_fet.cpp

```diff
@@ -0,0 +1,271 @@
+  if (given_quantity == "flux") {
+    quant = LegendreFET::Quantity::Flux;
+    found_quantity = true;
+  }
+
+  if (given_quantity == "fission") {
+    quant = LegendreFET::Quantity::Fission;
+    found_quantity = true;
+  }
+  if (given_quantity == "absorption") {
+    quant = LegendreFET::Quantity::Absorption;
+    found_quantity = true;
+  }
+  if (given_quantity == "elastic") {
+    quant = LegendreFET::Quantity::Elastic;
+    found_quantity = true;
+  }
+
+  if (found_quantity == false) {
+    fatal_error("For " + legendre_fet_tally_name +
+                " tally, a unknown quantity is given.");
+  }
+
+  // Get the enrgy bounds, if any is given
+  std::shared_ptr<EnergyFilter> energy_filter_ = nullptr;
+  if (node["energy-bounds"]) {
+    std::vector<double> energy_bounds =
+        node["energy-bounds"].as<std::vector<double>>();
+    energy_filter_ = std::make_shared<EnergyFilter>(energy_bounds);
+  }
+
+  // Get the cartesian type position filter
+  std::shared_ptr<CartesianFilter> cartesian_filter_ =
+      make_cartesian_filter(node);
+
+  // Get the Legendre-FET order
+  if (!node["FET-order"]) {
+    fatal_error("Legendre-FET order is not given for the " +
+                legendre_fet_tally_name + "tally.");
+  }
+  int FET_order_int_type = node["FET-order"].as<int>();
+  if (FET_order_int_type < 0) {
+    fatal_error("Legendre-FET order should not be negative.");
+  }
+  std::size_t FET_order_ = static_cast<std::size_t>(FET_order_int_type);
+
+  // Get the Legendre-FET axes
+  if (!node["FET-axis"]) {
+    fatal_error("The axes for " + legendre_fet_tally_name +
+                " should be given. ");
+  }
+  std::vector<std::string> fet_axis_type_string =
+      node["FET-axis"].as<std::vector<std::string>>();
+
+  std::vector<LegendreFET::Axis> fet_axis;
+  fet_axis.reserve(fet_axis_type_string.size());
+
+  for (auto& c : fet_axis_type_string) {
+    if (c == "X") fet_axis.push_back(LegendreFET::Axis::X);
+
+    if (c == "Y") fet_axis.push_back(LegendreFET::Axis::Y);
+
+    if (c == "Z") fet_axis.push_back(LegendreFET::Axis::Z);
+  }
+
+  if (!(fet_axis.size() == fet_axis_type_string.size())) {
+    fatal_error("A non-allowable name of the axis is given for " +
+                legendre_fet_tally_name + ".");
+  }
+
+  // Make the Legendre FET tally class
+  std::shared_ptr<LegendreFET> itally_legendre_fet_tally_ = nullptr;
+  if (estimator_name == "collision") {
+    itally_legendre_fet_tally_ = std::make_shared<LegendreFET>(
+        cartesian_filter_, energy_filter_, fet_axis, FET_order_, quant,
+        LegendreFET::Estimator::Collision, legendre_fet_tally_name);
+  }
+
+  if (estimator_name == "track-length") {
+    fatal_error(
+        "The track-length for the legendre-FET is not supported as asked in " +
+        legendre_fet_tally_name + " tally. ");
+  }
+
+  if (itally_legendre_fet_tally_ == nullptr)
+    fatal_error("Incorrect \"estimator\" is given.");
+
+  return itally_legendre_fet_tally_;
```

Comment:
Will also need to clean up this function, using chained if-else if -else and such.